pyswmm/output.py

```diff
@@ -980,4 +992,183 @@ def system_result(self, time_index: Union[int, datetime, None] = None):
         )
 
         values = output.get_system_result(self.handle, time_index, dummy_index)
-        return {attr: value for attr, value in zip(shared_enum.SystemAttribute, values)}
+        return {attr: value for attr, value in zip(
+            shared_enum.SystemAttribute, values)}
+
+
+class OutAttributeBase():
+    """Baseclass for SubcatchSeries, NodeSeries, LinkSeries."""
+
+    def __init__(self, out_handle):
+        if not isinstance(out_handle, Output):
+            raise (Exception("Invalid Outfile Handle"))
```

Comment:
Should raise a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError)